### ISSUE TITLE
Outdated documentation for argocd custom health check defintion

### ISSUE THREAD

**Issue by MattFanto** [argoproj/argo-cd#6719]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

I tried to implement an health check for a custom resource, as part of the argo-cm as decribed here https://argoproj.github.io/argo-cd/operator-manual/health/#way-1-define-a-custom-health-check-in-argocd-cm-configmap 
```
data:
  resource.customizations.health.cert-manager.io_Certificate: |
    hs = {}
    if obj.status ~= nil then
      if obj.status.conditions ~= nil then
        for i, condition in ipairs(obj.status.conditions) do
          if condition.type == "Ready" and condition.status == "False" then
            hs.status = "Degraded"
            hs.message = condition.message
            return hs
          end
          if condition.type == "Ready" and condition.status == "True" then
            hs.status = "Healthy"
            hs.message = condition.message
            return hs
          end
        end
      end
    end

    hs.status = "Progressing"
    hs.message = "Waiting for certificate"
    return hs

```
unfortuntately this approach didn't work and the Lua code was completely ignored by argo controller.

After few hours of troubleshooting I found this issue here https://github.com/argoproj/argo-cd/issues/3744 and I realized the documentation is outdated, the correct example should be:
```
data:
      resource.customizations: |-
        cert-manager.io/Certificate:
          health.lua: |
            hs = {}
            if obj.status ~= nil then
              if obj.status.conditions ~= nil then
                for i, condition in ipairs(obj.status.conditions) do
                  if condition.type == "Ready" and condition.status == "False" then
                    hs.status = "Degraded"
                    hs.message = condition.message
                    return hs
                  end
                  if condition.type == "Ready" and condition.status == "True" then
                    hs.status = "Healthy"
                    hs.message = condition.message
                    return hs
                  end
                end
              end
            end

            hs.status = "Progressing"
            hs.message = "Waiting for certificate"
            return hs

```

Using this format
```
      resource.customizations: |-
        <group_kind>/<CustomResource>:
          health.lua: |

```
Instead of 
```
      resource.customizations.health.<group_kind>_<CustomResource>: |
```
Solved my issue

**To Reproduce**

N.A.

**Expected behavior**

N.A.

**Screenshots**

N.A.

**Version**

2.0.4

**Logs**

N.A.

**Comment by mkasinat** [comment]
@MattFanto I have replaced my custom cert health as you defined but still cert health is not executing at all . Please see my config which is used .
cert-manager.io/Certificate:
      health.lua: |
        print ("LUA:CERT")
        hs = {}
        if obj.status ~= nil then
          if obj.status.conditions ~= nil then
            for i, condition in ipairs(obj.status.conditions) do
              if condition.type == "Ready" and condition.status == "False" then
                hs.status = "Degraded"
                hs.message = condition.message
                return hs
              end
              if condition.type == "Ready" and condition.status == "True" then
                hs.status = "Healthy"
                hs.message = condition.message
                return hs
              end
            end
          end
        end

        hs.status = "Progressing"
        hs.message = "Waiting for certificate"
        return hs
 bitnami.com/SealedSecret:
    health.lua: |
      hs = {}
      hs.status = "Healthy"
      hs.message = "Controller doesn't report resource status"
      return hs

do you any idea or anything wrong in my config?

**Comment by ThoTischner** [comment]
> @MattFanto I have replaced my custom cert health as you defined but still cert health is not executing at all . Please see my config which is used .
> cert-manager.io/Certificate:
> health.lua: |
> print ("LUA:CERT")
> hs = {}
> if obj.status ~= nil then
> if obj.status.conditions ~= nil then
> for i, condition in ipairs(obj.status.conditions) do
> if condition.type == "Ready" and condition.status == "False" then
> hs.status = "Degraded"
> hs.message = condition.message
> return hs
> end
> if condition.type == "Ready" and condition.status == "True" then
> hs.status = "Healthy"
> hs.message = condition.message
> return hs
> end
> end
> end
> end
> 
> ```
>     hs.status = "Progressing"
>     hs.message = "Waiting for certificate"
>     return hs
> ```
> 
> bitnami.com/SealedSecret:
> health.lua: |
> hs = {}
> hs.status = "Healthy"
> hs.message = "Controller doesn't report resource status"
> return hs
> 
> argocd cm config:
> resource.customizations: |2-
> cert-manager.io/Certificate:
> health.lua: |
> print ("LUA:CERT")
> hs = {}
> if obj.status ~= nil then
> if obj.status.conditions ~= nil then
> for i, condition in ipairs(obj.status.conditions) do
> if condition.type == "Ready" and condition.status == "False" then
> hs.status = "Degraded"
> hs.message = condition.message
> return hs
> end
> if condition.type == "Ready" and condition.status == "True" then
> hs.status = "Healthy"
> hs.message = condition.message
> return hs
> end
> end
> end
> end
> 
> ```
>         hs.status = "Progressing"
>         hs.message = "Waiting for certificate"
>         return hs
>  bitnami.com/SealedSecret:
>     health.lua: |
>       hs = {}
>       hs.status = "Healthy"
>       hs.message = "Controller doesn't report resource status"
>       return hs
> ```
> 
> do you any idea or anything wrong in my config?

Can you fix your code block please ?
Why the print ("LUA:CERT") ? Try to remove this line.

**Comment by mkasinat** [comment]
Yes I have removed the print statement and executed the code is it not at all considering even my certificate is unhealthy but still instance is showing as helathy only.i want the custom cert health suppose certificate is unhealthy I want my instance as unhealthy

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by mkasinat** [comment]
> > Yes I have removed the print statement and executed the code is it not at all considering even my certificate is unhealthy but still instance is showing as helathy only.i want the custom cert health suppose certificate is unhealthy I want my instance as unhealthy
> 
> Try to enable the openlibs: `resource.customizations.useOpenLibs.cert-manager.io_Certificate: "true"`

We are not able to enable the openlibs .

**Comment by caerulescens** [comment]
For anyone wondering the status of this ticket, the stable documentation is still wrong [here](https://argo-cd.readthedocs.io/en/stable/operator-manual/health/#way-1-define-a-custom-health-check-in-argocd-cm-configmap). I was having the same issue trying to get `cert-manager`'s certificates as healthy, and the stable documentation answer doesn't work. The search engines don't surface this GitHub ticket, so I had to sift through the issues to find this. I think a lot of people are having the same experience.
